Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+      ]
+    },
+    "ObjReproductionSpawner": {
+      "fields": [
+        { "name": "ReviveTime", "type": "float32" },
+        { "name": "Type", "type": "ObjReproductionSpawner::Type" }
+      ]
+    },
+    "ObjShootingSpawner": {
+      "fields": [
+        { "name": "MoveRate", "type": "float32" },
+        { "name": "MoveLength", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "MoveType", "type": "ObjShootingSpawner::MoveType" },
+        { "name": "ActionType", "type": "ObjShootingSpawner::ActionType" }
+      ]
+    },
+    "ObjVaultSpawner": {
+      "fields": [
+        { "name": "MoveSpeed", "type": "float32" },
+        { "name": "GuidePathID", "type": "uint32" },
+        { "name": "Path1DPos", "type": "float32" },
+        { "name": "ActionType", "type": "ObjVaultSpawner::ActionType" },
+        { "name": "OffsetType", "type": "ObjVaultSpawner::OffsetType" }
+      ]
+    },
+    "ObjBigCatapultSpawner": {
+      "fields": [
+        { "name": "UpSpeed", "type": "float32" },
+        { "name": "DownSpeed", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "StopTime", "type": "float32" },
+        { "name": "DeadPos", "type": "float32" },
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "Size", "type": "ObjBigCatapultSpawner::Size" },
+        { "name": "Mode", "type": "ObjBigCatapultSpawner::Mode" },
+        { "name": "IsDeadTypeCrush", "type": "bool" },
+        { "name": "IsMessageBoot", "type": "bool" }
+      ]
+    },
+    "ObjBilliardSpawner": {
+      "fields": [
+        { "name": "ModelType", "type": "ObjBilliardSpawner::ModelType" },
+        { "name": "NumMaxCollisions", "type": "uint8" }
+      ]
+    },
+    "EventDimChangetoLoopSpawner": {
+      "fields": [
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "CollisionHeight", "type": "float32" },
+        { "name": "m_PathEaseTime", "type": "float32" },
+        { "name": "TargetDirection", "type": "vector3" },
+        { "name": "m_IsChangeCamera", "type": "bool" },
+        { "name": "m_IsPadCorrect", "type": "bool" },
+        { "name": "m_IsAutoChange2DPath", "type": "bool" },
+        { "name": "m_IsEnableFromFront", "type": "bool" },
+        { "name": "m_IsEnableFromBack", "type": "bool" }
+      ]
+    },
+    "ObjEggmanMedalSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjEggMaskBoxSpawner": {
+      "fields": [
+        { "name": "AttackInterval", "type": "float32" },
+        { "name": "SearchRadius", "type": "float32" },
+        { "name": "DeadTime", "type": "float32" },
+        { "name": "OffsetY", "type": "float32" },
+        { "name": "OffsetX", "type": "float32" }
+      ]
+    },
+    "ObjEggMaskCollisionSpawner": {
+      "fields": [
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "CollisionHeight", "type": "float32" }
+      ]
+    },
```

Comment:
Parameters named based on in-game functionality and parameter names from other objects with similar functionality.